internal/processors/token_transfer.go

```diff
@@ -0,0 +1,199 @@
+	ledgerNumber := tx.Ledger.LedgerSequence()
+	txHash := tx.Hash.HexString()
+
+	// Get events from this specific transaction
+	txEvents, err := p.eventsProcessor.EventsFromTransaction(tx)
+	if err != nil {
+		return nil, fmt.Errorf("processing token transfer events from transaction: %w", err)
+	}
+
+	// Combine fee events and operation events
+	allEvents := append(txEvents.FeeEvents, txEvents.OperationEvents...)
+
+	stateChanges := make([]types.StateChange, 0, len(allEvents))
+	for _, e := range allEvents {
+		meta := e.GetMeta()
+		contractAddress := meta.GetContractAddress()
+
+		// Get operation details if available
+		var operationType *xdr.OperationType
+		var opSourceAccount string
+		if meta.GetOperationIndex() > 0 {
+			// GetOperation expects 0-based index, but meta has 1-based index
+			if op, found := tx.GetOperation(meta.GetOperationIndex() - 1); found {
+				operationType = &op.Body.Type
+				opSourceAccount = op.SourceAccount.Address()
+			}
+		}
+
+		baseChange := p.createBaseStateChange(ledgerNumber, ledgerCloseTime, txHash)
+		changes, err := p.processEventWithOperation(e.GetEvent(), contractAddress, baseChange, operationType, opSourceAccount)
+		if err != nil {
+			return nil, err
+		}
+		stateChanges = append(stateChanges, changes...)
+	}
+
+	return stateChanges, nil
+}
+
+func (p *TokenTransferProcessor) processEventWithOperation(event any, contractAddress string, baseChange types.StateChange, operationType *xdr.OperationType, opSourceAccount string) ([]types.StateChange, error) {
+	switch event := event.(type) {
+	case *ttp.TokenTransferEvent_Transfer:
+		return p.handleTransfer(event.Transfer, contractAddress, baseChange, operationType)
+	case *ttp.TokenTransferEvent_Mint:
+		return p.handleMint(event.Mint, contractAddress, baseChange)
+	case *ttp.TokenTransferEvent_Burn:
+		return p.handleBurn(event.Burn, contractAddress, baseChange)
+	case *ttp.TokenTransferEvent_Clawback:
+		return p.handleClawback(event.Clawback, contractAddress, baseChange, operationType, opSourceAccount)
+	case *ttp.TokenTransferEvent_Fee:
+		return p.handleFee(event.Fee, contractAddress, baseChange)
+	default:
+		return nil, fmt.Errorf("unknown event type: %T", event)
+	}
+}
+
+func (p *TokenTransferProcessor) createBaseStateChange(ledgerNumber uint32, ledgerCloseTime int64, txHash string) types.StateChange {
+	change := types.StateChange{
+		LedgerNumber:    int64(ledgerNumber),
+		LedgerCreatedAt: time.Unix(ledgerCloseTime, 0),
+		IngestedAt:      time.Now(),
+		TxHash:          txHash,
+	}
+
+	return change
+}
+
+func (p *TokenTransferProcessor) setAssetOrContract(change *types.StateChange, asset *asset.Asset, contractAddress string) {
+	if asset != nil {
+		if asset.GetNative() {
+			change.Token = sql.NullString{String: "native", Valid: true}
+		} else if issuedAsset := asset.GetIssuedAsset(); issuedAsset != nil {
+			change.Token = sql.NullString{String: fmt.Sprintf("%s:%s", issuedAsset.GetAssetCode(), issuedAsset.GetIssuer()), Valid: true}
```

Comment:
FYI: we have a constructor for NullString, so you con't need to set valid=true:

https://github.com/stellar/wallet-backend/blob/8be19279f16c509e89762dfd1604cc711d465b59/internal/utils/sql.go#L8-L20